RealmBrowser/Classes/RLMRealmBrowserWindowController.m

```diff
@@ -101,6 +108,80 @@ - (IBAction)saveObjcModels:(id)sender
     [RLMModelExporter saveModelsForSchemas:objectSchemas inLanguage:RLMModelExporterLanguageObjectiveC];
 }
 
+- (IBAction)saveCopy:(id)sender
+{
+    NSString *fileName = [self.modelDocument.presentedRealm.realm.path lastPathComponent];
+    NSSavePanel *panel = [NSSavePanel savePanel];
+    [panel setNameFieldStringValue:fileName];
+    [panel beginSheetModalForWindow:self.window completionHandler:^(NSInteger result){
+        if (result != NSFileHandlingPanelOKButton)
+            return;
+        
+        dispatch_async(dispatch_get_main_queue(), ^{
+            NSURL *fileURL = [panel URL];
+            [self exportAndCompactCopyOfRealmFileAtURL:fileURL];
+        });
+    }];
+}
+
+- (void)exportAndCompactCopyOfRealmFileAtURL:(NSURL *)realmFileURL
+{
+    NSError *error = nil;
+    
+    //Check that this won't end up overwriting the original file
+    if ([realmFileURL.path.lowercaseString isEqualToString:self.modelDocument.presentedRealm.realm.path.lowercaseString]) {
+        NSAlert *alert = [[NSAlert alloc] init];
+        alert.messageText = @"You cannot overwrite the original Realm file.";
+        alert.informativeText = @"Please choose a different location in which to save this Realm file.";
+        [alert runModal];
+        return;
+    }
+    
+    //Ensure a file with the same name doesn't already exist
+    BOOL directory = NO;
+    
+    if ([[NSFileManager defaultManager] fileExistsAtPath:realmFileURL.path isDirectory:&directory] && !directory) {
```

Comment:
isn't it common to prompt before overwriting a file? I don't know if NSSavePanel would ever return a directory here, but if it does, this code will skip this deletion step and attempt to write a realm file to a location matching a directory, which will throw an exception.